### PR TITLE
fix(provided-externals): logs warning for child modules

### PR DESCRIPTION
**Pull request by Francois-Esquire**
Replaced error thrown `onModuleLoad` with warning log for child modules that contain the `providedExternals` in `appConfig`.

### CODE REVIEW

File: prod-sample/sample-modules/late-frank/0.0.1/src/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import ModuleContainer from './components/LateFrank';
+
+if (!global.BROWSER) {
+  ModuleContainer.appConfig = {
```
Comment (10xLaCroixDrinker) [inline prod-sample/sample-modules/late-frank/0.0.1/src/index.js]: this should be done by the bundler

**Comment by Francois-Esquire** [comment]
☑️ Preflight Checklist:

What is the communication plan for this change?
None, `providedExternals` pending documentation

Does any documentation need to be updated or added to account for this change? If so was it done already?
No

What is the motivation for this change?
To prevent the app from crashing based on child module mis-configuration

Should these changes also be applied to a maintenance branch or any other long lived branch?
No

How was this change tested?
Unit and integration testing

Does this change require cross browser checks? Why or why not?
No, server side behavior

Does this change require a performance test prior to merging? Why or why not?
No, replaced throwing an error with `console.warn`

Could this be considered a breaking change? Why or why not?
No, API is unchanged

Does the change impact caching?
No

Does the change impact HTTP headers?
No

Does the change have any new infrastructure requirements?
No

Does the change affect other versions of the app?
No

Does the change require additional environment variables?
No

What is the impact to tenants?
None

What is the impact to individual users?
Warning on mis-configuration of `appConfig`

What is the change to the size of assets?
None

Should integration tests be added to protect against future regressions on this change?
Added

File: __tests__/integration/one-app.spec.js [hunk 2]
```diff
@@ -568,6 +568,41 @@ describe('Tests that require Docker setup', () => {
...
+          // not ideal but need to wait for app to poll;
+          await waitFor(5000);
+        });
+
+        afterAll(() => {
+          writeModuleMap(originalModuleMap);
+        });
+
+        test(
+          'writes a warning to log if a child module has set `providedExternals` in `appConfig`',
```
Comment (10xLaCroixDrinker) [inline __tests__/integration/one-app.spec.js]: you need to test that the externals provided by the root are still correct/available and that the ones provided by the non-root module are not available

File: src/server/utils/onModuleLoad.js [hunk 3]
```diff
@@ -118,7 +118,9 @@ export default function onModuleLoad({
   }
 
   if (providedExternals) {
-    throw new Error(`Module ${moduleName} attempted to provide externals, but it is not the root module.`);
+    console.warn(
+      `Module ${moduleName} attempted to provide externals. Only the root module can provide externals.`
+    );
```
Comment (10xLaCroixDrinker) [inline src/server/utils/onModuleLoad.js]: You will need some changes around here: https://github.com/americanexpress/one-app-cli/blob/master/packages/one-app-bundler/webpack/loaders/provided-externals-loader.js#L34

File: __tests__/integration/one-app.spec.js (hunk 2 above)
Comment (Francois-Esquire) [reply __tests__/integration/one-app.spec.js]: Got it, I'll update the tests

File: prod-sample/sample-modules/late-frank/0.0.1/src/index.js (hunk 1 above)
Comment (Francois-Esquire) [reply prod-sample/sample-modules/late-frank/0.0.1/src/index.js]: What do you mean by this specifically? The configuration?

Comment (Francois-Esquire) [reply prod-sample/sample-modules/late-frank/0.0.1/src/index.js]: I see what you mean now, will update.

File: src/server/utils/onModuleLoad.js (hunk 3 above)
Comment (Francois-Esquire) [reply src/server/utils/onModuleLoad.js]: I think it's best we move that helper added by the loader into One App -> `getRootModule` and remove it from being injected. Thoughts?

Comment (mtomcal) [reply src/server/utils/onModuleLoad.js]: - [ ] Get @10xLaCroixDrinker Feedback on this

File: __tests__/integration/one-app.spec.js (hunk 2 above)
Comment (Francois-Esquire) [reply __tests__/integration/one-app.spec.js]: updated aa0b3a1

File: prod-sample/sample-modules/late-frank/0.0.1/src/index.js (hunk 1 above)
Comment (Francois-Esquire) [reply prod-sample/sample-modules/late-frank/0.0.1/src/index.js]: updated aa0b3a1

**Pull request merged**